lms/djangoapps/instructor/views/api.py

```diff
@@ -2721,20 +2840,31 @@ def create_certificate_exception(request, course_id, white_list_student=None):
             'success': False,
             'message': _('Invalid Json data')
         }, status=400)
-    try:
-        certificate_white_list, students = process_certificate_exceptions(certificate_white_list, course_key)
-    except ValueError as error:
+
+    users = [exception.get('user_id', False) for exception in certificate_white_list]
```

Comment:
I think when `generate_for == 'all'` we should not pass list of users since we are getting those from DB. In that case this check does not make sense.